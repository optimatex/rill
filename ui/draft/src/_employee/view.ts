export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
