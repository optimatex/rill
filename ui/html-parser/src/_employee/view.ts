export { React, RC } from '@rill/view';
