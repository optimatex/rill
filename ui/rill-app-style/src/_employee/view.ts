export { createGlobalStyle, css } from '@rill/view';
