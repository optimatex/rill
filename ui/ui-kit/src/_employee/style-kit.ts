export {
  // appLayout
  appLayoutCss,
  screenCss,
  sidebarColumnCss,
  flatColumnCss,
  mainColumnCss,
  // layout
  listCss,
  // Brand
  logCss,
  dualLogoCss,
  // Controller
  trivialAppHeaderCss,
  // _menu
  menuWrapCss,
  menuItemCss,
  // Image
  imageCss,
  imagesListCss,
  imagesGridCss,
  imagesLayoutItemCss,
  imageCropperCss,
  // Tags
  tag,
  tagsList,
  // Ui-primal
  // _grid
  gridRowContainerCss,
  gridrillRowContainerCss,
  gridColumnCss,
  // _card
  cardContainerCss,
  postContainerCss,
  postTitleCss,
  panelCss,
  // _buttons
  buttonCss,
  buttonActiveCss,
  buticonCss,
  // _form
  formContainerCss,
  formFieldFlatCss,
  formFieldColumnCss,
  formFieldErrorCss,
  formLabelTextCss,
  inputCss,
  toggleContainerCss,
  toggleContainerActiveCss,
  toggleRoundCss,
  toggleRoundActiveCss,
  // _article
  paragraphCss,
  nameCss,
  richContentWrapCss,
  // _table
  tableContainerCss,
  tableHeadCss,
  tableRowCss,
  tableColumnCss,
  // Unknown
  // _object
  stringValueCss,
  numberValueCss,
  booleanValueCss,
  emptyValueCss,
  arrayWrapCss,
  arrayItemCss,
  objectKeyCss,
  objectWrapCss,
  objectPairCss,
  // Unsorted
  lightIndicatorCss,
  popoverCss,
  pipeCss,
  hintableLogCss,
  fixedPanelCss,
} from '@rill/style-kit';
