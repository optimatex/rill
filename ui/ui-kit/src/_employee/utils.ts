export {
  toPairs,
  find,
  has,
  path,
  prop,
  curry,
  assoc,
  lensProp,
  view,
  set,
} from '@rill/utils';
