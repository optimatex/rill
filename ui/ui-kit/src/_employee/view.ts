export {
  React,
  useState,
  useCallback,
  // styled-components
  styled,
  // Lens
  // getTestId,
  // Types
  FC,
  RC,
  ReactNode,
  LegacyRef,
  // PropTypes
  IContainerProps,
  // Event types
  FormEvent,
  MouseEvent,
} from '@rill/view';
