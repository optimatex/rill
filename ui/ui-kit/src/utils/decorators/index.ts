export * from './defaultProps';
export * from './testIdOperations';
export * from './testId';
