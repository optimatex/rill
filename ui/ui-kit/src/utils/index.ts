export * from './menu';
export * from './decorators';
