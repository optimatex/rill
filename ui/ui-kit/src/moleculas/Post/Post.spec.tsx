import { React } from '_employee/view';
import { render, text_1, cardSR } from 'test';
import { postIdKey } from 'utils';
import { Post } from '.';

const props_1 = {
  title: text_1,
};

const draw = (props = {}) =>
  render(
    <Post {...props}>
      <button>button text</button>
    </Post>,
  );

describe('Post', () => {
  it('should render without errors', () => {
    const { getByTestId } = draw(props_1);
    const el = getByTestId(postIdKey);

    expect(el).toMatchSnapshot();
  });

  it('should render post title', () => {
    const { getByRole } = draw(props_1);
    const el = getByRole('heading');

    expect(el).toMatchSnapshot();
  });

  it('should render post body', () => {
    const { getByRole } = draw();
    const el = getByRole('button');

    expect(el).toMatchSnapshot();
  });

  it('card body should contains all style rules', () => {
    const { getByTestId } = draw();
    const el = getByTestId(postIdKey);

    expect(el).toHaveStyleRule('background', cardSR.background);
  });
});

// it('card title should contains all style rules', () => {
//   const { getByRole } = draw();
//   const el = getByRole('heading');

//   expect(el).toHaveStyleRule('font-size', fontSR.fontSize);
// });

// it('card title should contains all style rules', () => {
//   const { getByRole } = draw();
//   const el = getByRole('heading');

//   expect(el).toHaveStyleRule('font-size', fontSR.fontSize);
// });

// it('card body should contains all style rules', () => {
//   const { getByText } = draw_content_card();
//   const el = getByText(text_2);

//   expect(el).toHaveStyleRule('background', cardSR.background);
// });
