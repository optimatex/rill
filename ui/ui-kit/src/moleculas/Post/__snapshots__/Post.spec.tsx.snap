// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render post body 1`] = `
<button>
  button text
</button>
`;

exports[`Post should render post title 1`] = `
.c0 {
  margin: 0;
  line-height: 1rem;
  font-family: 'open-sans-regular';
}

<h3
  class="c0"
>
  1 Some content 1
</h3>
`;

exports[`Post should render without errors 1`] = `
.c2 {
  margin: 0;
  line-height: 1rem;
  font-family: 'open-sans-regular';
}

.c0 {
  padding: 0;
  border-width: 1px;
  border-style: solid;
  background: red;
  color: white;
}

.c0 > *:nth-child(1) {
  border-radius: 4px 4px 0 0;
}

.c0 > *:last-child {
  border-radius: 0 0 4px 4px;
}

.c1 {
  box-sizing: border-box;
  padding: 5px 8px;
  border-bottom-width: 1px;
  border-bottom-style: solid;
}

<div
  class="c0"
  data-testid="post"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      1 Some content 1
    </h3>
  </div>
  <button>
    button text
  </button>
</div>
`;
